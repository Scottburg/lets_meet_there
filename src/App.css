.App {
  text-align: center;
  width: 100%;
}

.App-logo {
  height: 30vmin;
  pointer-events: none;
  margin-top: 30px;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 30s linear;
  }
}

/* .Currency-buttons {
  display: flex;
  justify-content: flex-end;
  margin-right: 20px;
  color: rgba(255, 72, 0, 0.789);
} */
.selected {
  text-decoration: underline;
}

/* .App-header {
  background-color: #282c34;
  min-height: 10vh;
  width: 90vw;
  display: flex;
  flex-direction: column;
  align-items: right;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  margin: 20px;
} */
.App-body {
  background-color: #282c34;
  min-height: 130vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
