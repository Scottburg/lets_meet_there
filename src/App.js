import React, { useState } from 'react';
import logo from './assets/Plane.svg';
import ApiClient from './Services/ApiClient';
import './App.css';
import SearchForm from './Components/SearchForm/searchForm';
import FlightList from './Containers/FlightList/flightList';
import helpers from './helpers';
import { useSelector, useDispatch } from 'react-redux';
import { isLoading, getPlaces, getCarriers } from './Actions';
function App() {
  const [matched, setMatched] = useState([]);
  // Redux items
  const loading = useSelector((state) => state.isLoading);
  const dispatch = useDispatch();

  const getPlace = async (query) => {
    return ApiClient.getPlace(query).then((res) => {
      return helpers.placeId(res, query);
    });
  };

  const searchFlights = async (from1, from2, departDate, returnDate) => {
    dispatch(isLoading());
    const quotesA = await ApiClient.getFlights(from1, departDate, returnDate);
    const quotesB = await ApiClient.getFlights(from2, departDate, returnDate);
    dispatch(getPlaces(quotesA.places, quotesB.places)); // dispatch is a redux function that gets the named reducer and sets the state.
    dispatch(getCarriers(quotesA.carriers, quotesB.carriers));
    setMatched(helpers.matchFlights(quotesA, quotesB)); // as this is passed through props fine to leave outside redux.
    dispatch(isLoading());
  };

  return (
    <div className="App-body">
      <div className="App">
        <header className="App-header"></header>
        <h1>Search for a place to meet</h1>

        <SearchForm
          searchFlights={searchFlights}
          getPlace={getPlace}
        ></SearchForm>
        <div>
          {loading ? (
            <img src={logo} className="App-logo" alt="logo" />
          ) : (
            <FlightList matchedFlights={matched}></FlightList>
          )}
        </div>
      </div>
    </div>
  );
}

export default App;
