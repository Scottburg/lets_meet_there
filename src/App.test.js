import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import App from './App';

const mockStore = configureStore([]);
const store = mockStore();

describe('test the Spinner', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(asFragment(<App />)).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const { getByText } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const title = getByText('Search for a place to meet');
    expect(title).toBeInTheDocument();
  });
});
