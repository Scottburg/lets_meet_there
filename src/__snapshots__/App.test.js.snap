// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the Spinner should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App-body"
  >
    <div
      class="App"
    >
      <header
        class="App-header"
      />
      <h1>
        Search for a place to meet
      </h1>
      <div
        class="form-container"
      >
        <div
          class="form"
        >
          <form
            class="form"
          >
            <div
              class="from1"
            >
              <div
                class="formTitles"
              >
                From
              </div>
              <label
                for="from1"
              />
              <input
                class="inputboxes"
                name="from1"
                placeholder="E.g. London"
                type="text"
                value=""
              />
            </div>
            <div
              class="from2"
            >
              <div
                class="formTitles"
              >
                And
              </div>
              <label
                for="from2"
              />
              <input
                class="inputboxes"
                name="from2"
                placeholder="E.g. Barcelona"
                type="text"
                value=""
              />
            </div>
            <div
              class="datePicker"
            >
              <div
                class="DateRangePicker DateRangePicker_1"
              >
                <div>
                  <div
                    class="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2 DateRangePickerInput__showClearDates DateRangePickerInput__showClearDates_3"
                  >
                    <div
                      class="DateInput DateInput_1"
                    >
                      <input
                        aria-describedby="DateInput__screen-reader-message-startDate"
                        aria-label="Depart"
                        autocomplete="off"
                        class="DateInput_input DateInput_input_1"
                        id="startDate"
                        name="startDate"
                        placeholder="Depart"
                        type="text"
                        value=""
                      />
                      <p
                        class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                        id="DateInput__screen-reader-message-startDate"
                      >
                        Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                      </p>
                    </div>
                    <div
                      aria-hidden="true"
                      class="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
                      role="presentation"
                    >
                      <svg
                        class="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
                        focusable="false"
                        viewBox="0 0 1000 1000"
                      >
                        <path
                          d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
                        />
                      </svg>
                    </div>
                    <div
                      class="DateInput DateInput_1"
                    >
                      <input
                        aria-describedby="DateInput__screen-reader-message-endDate"
                        aria-label="Return"
                        autocomplete="off"
                        class="DateInput_input DateInput_input_1"
                        id="endDate"
                        name="endDate"
                        placeholder="Return"
                        type="text"
                        value=""
                      />
                      <p
                        class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                        id="DateInput__screen-reader-message-endDate"
                      >
                        Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                      </p>
                    </div>
                    <button
                      aria-label="Clear Dates"
                      class="DateRangePickerInput_clearDates DateRangePickerInput_clearDates_1 DateRangePickerInput_clearDates_default DateRangePickerInput_clearDates_default_2 DateRangePickerInput_clearDates__hide DateRangePickerInput_clearDates__hide_3"
                      type="button"
                    >
                      <svg
                        class="DateRangePickerInput_clearDates_svg DateRangePickerInput_clearDates_svg_1"
                        focusable="false"
                        viewBox="0 0 12 12"
                      >
                        <path
                          d="M11.53.47a.75.75 0 0 0-1.061 0l-4.47 4.47L1.529.47A.75.75 0 1 0 .468 1.531l4.47 4.47-4.47 4.47a.75.75 0 1 0 1.061 1.061l4.47-4.47 4.47 4.47a.75.75 0 1 0 1.061-1.061l-4.47-4.47 4.47-4.47a.75.75 0 0 0 0-1.061z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="buttonDiv"
            >
              <button
                type="submit"
              >
                 Lets Meet There!
              </button>
            </div>
          </form>
        </div>
      </div>
      <div>
        <div>
          <div
            class="tile_Container"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
